src/Microsoft.AspNet.Razor/Parser/TagHelpers/TagHelperBlockRewriter.cs

```diff
@@ -363,6 +387,35 @@ private static Block RebuildCodeGenerators(Block block)
             return builder.Build();
         }
 
+        private static SourceLocation GetAttributeNameStartLocation(SyntaxTreeNode node)
+        {
+            var startOffset = node.Start;
```

Comment:
name looks wrong. this appears to be the `blockStart` and `firstNonWhitespaceSymbol.Start` provides the offset from that location.